### DISCUSSION TITLE
How to keep navigation pane with narrower windows?

### DISCUSSION THREAD

**Discussion by hsutter** [squidfunk/mkdocs-material#6785] [Q&A] [answered]
Is there a way to make the left navigation pane stay visible at slightly higher browser Zoom levels / narrower browser windows?

This screenshot from my documentation shows the left navigation pane:

<p align="center"><img width="600" src="https://github.com/hsutter/cppfront/assets/1801526/0cb72cc4-45b3-4464-bed4-e08a1a2c3e0c"></p>

... but when I hit the browser's `Ctrl`-`+` (zoom in) just one more time (because the text is a bit small) or narrow the window slightly, the navigation pane disappears:

<p align="center"><img width="600" src="https://github.com/squidfunk/mkdocs-material/assets/1801526/5599bbc5-2026-4084-aa32-35afc871f7dc"></p>

It seems like there's still plenty of room for the pane, given all the empty space / long text lines on the right half of the screenshot? Losing that navigation pane is sad, because it's so super useful. Without it, I quickly lose track of where I am in my own documentation.

I looked in the docs but the `navigation.*` options I tried didn't seem to affect it.

Update: I suspect the answer is related to the `1220px` limit value. Perhaps it's worth considering changing that to `1024px` or even `900px`? I encountered this problem using a vertical monitor, where the default browser 100% zoom level triggers the limit to hide the navigation, so it seems the navigation is invisible *by default* on at least some vertical monitors. Even though the text is pretty small (and lines pretty long) at that default zoom, I had to `Ctrl` `-` to make the navigation visible, which makes text even smaller (and lines even longer).

**Comment by alexvoss** [comment]
You are right that this is related to the break points defined in [_config.scss](https://github.com/squidfunk/mkdocs-material/blob/55603a70e843a8b5e3a4a2ac80c51435c2533600/src/templates/assets/stylesheets/_config.scss#L37). If I understand things correctly, this is not that easy to change since this value is fed into calculations that make the behaviour dependent on the font size, which is what you are observing when you change that using your browser's shortcuts.

If you wanted to change this, the best way would be to compile the theme yourself and adjust the 1220px value in the config file. You would need to fix your version or maintain this going forward, though. Your call if that would be worth it.

Doing this after the CSS has been compiled is somewhat difficult and fragile. I have seen @kamilkrzyskow [answer a similar question](https://github.com/squidfunk/mkdocs-material/discussions/6639#discussioncomment-8116000) by digging into the CSS in the browser developer tools. Overriding like this seems fragile, though.

As to whether this could be made configurable in the future, I am not sure. The theme is already configurable in many ways and the answer when something can be done by customization is often that it should be done that way by the user so as to not bloat the code and documentation for every conceivable adjustment. In this case, however, there may not be an easy way to customize. I will need to tag @squidfunk here to weigh in and tell us that we are holding the wrong end of the stick or to opine on whether there is a chance this could be made configurable. The way that the SCSS gets compiled into CSS does seem to make this somewhat unlikely but then I am not familiar enough to give a definitive answer.

I would also say that changing this may cause problems elsewhere. You may not be happy on your tilted monitor but would the result work for someone with a different device? There are so many combinations of size and resolution out there. I travel a fair bit, so have a 13" display with 2K resolution that I use in portrait mode - a perfect way to highlight the imperfections of our technical world... Just my 2p.

**Reply by squidfunk** [reply, marked as answer]
> As to whether this could be made configurable in the future, I am not sure. 

Once [`@custom-media`](https://www.w3.org/TR/mediaqueries-5/#custom-mq) is supported, we can make it configurable. Until then, the theme needs to be recompiled. Please understand that we cannot _just_ change this value. Tens of thousands of sites would be impacted. Note that on tablet, normally, the table of contents occupies the space on the right, but since you're using table of contents integration, it appears to be even wider. However, we cannot make the [significantly complex navigation CSS](https://github.com/squidfunk/mkdocs-material/blob/master/src/templates/assets/stylesheets/main/components/_nav.scss) any more complex, adding further exceptions for specific features.

The fact that we compile all of this from SCSS makes it adjustable in the first place. If we'd just use CSS, you would have to adjust every single instance. Since we compile SCSS, you can adjust the following values to fit and recompile:

https://github.com/squidfunk/mkdocs-material/blob/55603a70e843a8b5e3a4a2ac80c51435c2533600/src/templates/assets/stylesheets/_config.scss#L27-L42

Note that some further adjustments to the JavaScript source might be necessary as well.

**Reply by hsutter** [reply]
Thanks for both answers! I can only mark one as the answer but both are helpful.

> since you're using table of contents integration

Thanks, disabling that has partly mitigated my immediate problem — now the ToC shows on the right at the default zoom/width on a vertical monitor, though it's not quite as pretty as it was when integrated in the navigation. (And the ToC still disappears too quickly on zoom/narrowing, when body text lines are still avg. 19 words long, so I'd still love a configuration option when `@custom-media` is supported. But it disappears less quickly than when integrated in navigation.)

Thanks again for this great project.
